extensions/graphql/README.md

```diff
@@ -0,0 +1,221 @@
+# @loopback/graphql
+
+This module provides integration with [GraphQL](https://graphql.org/) using
+[type-graphql](https://typegraphql.com/).
+
+![type-graphql](type-graphql.png)
+
+## Stability: ⚠️Experimental⚠️
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+## Basic Use
+
+```ts
+export class MyApplication extends BootMixin(RestApplication) {
+  constructor(config: ApplicationConfig) {
+    super(config);
+    this.projectRoot = __dirname;
+    this.component(GraphQLComponent);
+    this.configure(GraphQLBindings.GRAPHQL_SERVER).to({asMiddlewareOnly: true});
+  }
+}
+```
+
+## Configure GraphQLServer
+
+This package can be used in two flavors:
+
+- As a server for LoopBack applications
+
+```ts
+import {Application} from '@loopback/core';
+import {GraphQLServer} from '@loopback/graphql';
+
+const app = new Application();
+const serverBinding = app.server(GraphQLServer);
+app.configure(serverBinding.key).to({host: '127.0.0.1', port: 0});
+server = await app.getServer(GraphQLServer);
+// ...
+await app.start();
+```
+
+- As a middleware for LoopBack REST applications
+
+```ts
+export class GraphqlTestApplication extends BootMixin(
+  RepositoryMixin(RestApplication),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.component(GraphQLComponent);
+    const server = this.getSync(GraphQLBindings.GRAPHQL_SERVER);
+    this.expressMiddleware('middleware.express.GraphQL', server.expressApp);
+    this.configure(GraphQLBindings.GRAPHQL_SERVER).to({
+      asMiddlewareOnly: true,
+    });
+
+    // ...
+
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {
+      graphqlResolvers: {
+        // Customize ControllerBooter Conventions here
+        dirs: ['graphql-resolvers'],
+        extensions: ['.js'],
+        nested: true,
+      },
+    };
+  }
+}
+```
+
+## Add a resolver
+
+Please add `recipe-resolver.ts` to `src/graphql-resolvers`.
+
+```ts
+@resolver(of => Recipe)
+export class RecipeResolver implements ResolverInterface<Recipe> {
```

Comment:
Please add `import` statements to show where are `ResolverInterface`, `@resolver` and `Recipe` coming from.